src/modules/apps/link.ts

```diff
@@ -29,7 +31,71 @@ const UPDATE_SIGN = chalk.blue('U')
+
+const injectedDependencies = async (builderName: string, builderVersion: string, workspace: string, account: string, environment: string) => {
+  const extension = (environment === 'prod') ? 'myvtex' : 'myvtexdev'  // Remove this
+  const http = axios.create({
+    baseURL: `https://${workspace}--${account}.${extension}.com`, // change this route!!!
+    timeout: builderHubInjectedDepsTimeout,
+  })
+  try {
+    const res = await http.get(`/_v/private/builder/0/getdeps/${builderName}/${builderVersion}`)
+    return res.data
+  } catch (e) {
+    return {}
+  }
+}
 
+const appTypingsURL = async (account: string, workspace: string, environment: string, appName: string, appVersion: string, builder: string): Promise<string> => {
+  const extension = (environment === 'prod') ? 'myvtex' : 'myvtexdev'  // Remove this
+  const appId = await resolveAppId(appName, appVersion)
+  const typingsPath = /\+build/.test(appId) ? 'linked/v1' : 'v1'
+  return `https://${workspace}--${account}.${extension}.com/_v/private/typings/${typingsPath}/${appId}/${builder}` // change this route!!!!!!
+}
+
+const appsWithTypingsURLs = async (builder: string, account: string, workspace: string, environment: string, appDependencies: {[key: string]: string}) => {
+  const result: {[key: string]: string} = {}
+  for (const [appName, appVersion] of Object.entries(appDependencies)) {
+    result[appName] = await appTypingsURL(account, workspace, environment, appName, appVersion, builder)
+  }
+  return result
+}
+
+const mergeDevDeps = (currentPackageJson: {[key: string]: any}, newDevDepsEntries: {[key: string]: string}) =>
+  merge(currentPackageJson.devDependencies, newDevDepsEntries)
+
+const runYarn = () => {
+  log.info('Running yarn in ./react/')
+  process.chdir('./react')
+  execSync(yarn, {stdio: 'inherit'})
+  process.chdir('../')
+  log.info('Finished running yarn')
+}
+
+const getTypings = async (builder: string, manifest: Manifest, account: string, workspace: string, environment: string) => {
+  const packageJsonPath = resolvePackageJsonPath(builder)
+  if (existsSync(packageJsonPath)) {
+    if (has(builder, manifest.builders)) {
+      const declaredDependencies = manifest.dependencies
+      const implicitDependencies = await injectedDependencies(builder, manifest.builders[builder], workspace, account, environment)
+      const appDependencies = merge(declaredDependencies, implicitDependencies)
+      if (appDependencies) {
+        log.info('Exporting app dependencies to react/package.json')
+        const reactPackageJson = JSON.parse(readFileSync(packageJsonPath, 'utf8'))
+        const newDevDepsEntries = await appsWithTypingsURLs(builder, account, workspace, environment, appDependencies)
+        reactPackageJson.devDependencies = mergeDevDeps(reactPackageJson, newDevDepsEntries)
+        writeFileSync(packageJsonPath, JSON.stringify(reactPackageJson, null, 2))
+        runYarn()
```

Comment:
Maybe I got it wrong, but it seems that every time the user runs `vtex link`, we are running `yarn install --force` locally, before sending the files to `builder-hub`, right? If this is the case, it is worth noting that this adds some overhead to every link request (and if the user, for some reason, does not use `yarn`, the first `vtex link` will take a long time).

Reply:
This is now changed. `yarn install --force` is only called if the content of `devDependencies` in `package.json` changes.